keps/sig-auth/740-service-account-external-signing/README.md

```diff
@@ -1,113 +1,560 @@
 
-### Risks and Mitigations
+##### Prerequisite testing updates
+
+##### Unit tests
+
+Will need coverage for:
+
+- pkg/kubeapiserver/options/authentication.go
+- the signer plugin 
+- the key cache plugin
+
+<!-- TODO
+Add info on concerned test in below form as read from https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+- `<package>`: `<date>` - `<test coverage>`
+ -->
+
+##### Integration tests
+
+- Create a cluster with ExternalJWTSigner to configure and verify the health status of core components as published by kube-apiserver.
+
+##### e2e tests
+
+- Create a cluster with ExternalJWTSigner configured.
+- Request a token for a service account principal.
+- Use a token as bearer for making requests to kube-apiserver and ensure it succeeds.
+
+### Graduation Criteria
+
+#### Alpha
+
+- Unit tests are completed
+- Initial integration tests are completed with a dummy ExternalSigner.
+
+#### Beta
+
+- E2E tests are completed.
+- We have at least one ExternalSigner implementation working with this change.
+
+#### GA
+
+- More than one ExternalSigner implementations are completed.
+- Feature is tuned with feedback from distributions.
+
+### Upgrade/Downgrade Strategy
+
+Existing clusters preserving existing functionality: 
+- Need no changes.
+
+Existing clusters using enhanced functionality: 
+- Need a MasterVM workload that implements ExternalJWTSigner
+- Need to configure the cluster with the `--service-account-signing-endpoint` flag. 
+
+
+### Version Skew Strategy
+
+The change will completely live in the Control plane and thus, there is no question of a skew with component interaction. The only potential for a skew arises with different kube-apiserver versions running at the same time
+
+#### Different kube-apiserver versions running at the same time
+- Skew is likely to exist when externalSigning is enabled/disabled in a live cluster.
+- Distributions can take one of 2 approaches to handle it:
+  1. Tolerate disruption
+    - Disruption would exist until all kube-apiserver instances have not restarted with the new config (`--service-account-signing-endpoint` being set/unset).
+    - During this, tokens can be issued by kube-apiserver running older version and fail auth when calling kube-apiserver running newer version.
+    - The opposite is also true.
+  2. Include old keys in supported set
+    - This is for distributions that are looking for a smoother transition when enabling/disabling external singing on live clusters.
+    - When disabling, distributions shall import keys from external signers to the on-disk key set.
+    - When enabling, distributions shall implement an external signer such that it combines the external keys with on-disk keys. 
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+For distributions that require capability to enable the feature on a live cluster:
```

Comment:
Need to mention setting the feature flag `ExternalServiceAccountTokenSigner` in the kube-apiserver (KAS)

Reply:
Done